### PR TITLE
Memory leak fix in the subscribers cache

### PR DESCRIPTION
**Pull request by nevesmarcio**
Signed-off-by: nevesmarcio nevesmarcio@gmail.com

### CODE REVIEW

**Comment by purplefox** [comment]
Looks good, but can you just fix your indentation?

**Comment by nevesmarcio** [comment]
I thought I could quickly do it with the online editor, but apparently not :)

File: vertx-hazelcast/src/main/java/org/vertx/java/spi/cluster/impl/hazelcast/HazelcastAsyncMultiMap.java [hunk 1]
```diff
@@ -114,11 +114,15 @@ public void handle(AsyncResult<Collection<V>> result) {
...
-              ChoosableSet<V> prev = cache.putIfAbsent(k, sids);
-              if (prev != null) {
-                // Merge them
-                prev.merge(sids);
-                sids = prev;
+              //SBKD-322 Quick fix for memory leak that has been caused by adding to subscribers cache
+              //empty sets. When no subscribers are present just omit caching it.
+              if (!sids.isEmpty()) {
+                ChoosableSet<V> prev = cache.putIfAbsent(k, sids);
+            	if (prev != null) {
```
Comment (purplefox) [inline vertx-hazelcast/src/main/java/org/vertx/java/spi/cluster/impl/hazelcast/HazelcastAsyncMultiMap.java]: We use 2 spaces for indentation.

**Comment by nevesmarcio** [comment]
Done (for real this time!)

**Comment by cescoffier** [comment]
Looks good now.

**Comment by pmlopes** [comment]
:+1:

**Comment by vietj** [comment]
merging this even if we won't release 2.x anymore

**Pull request merged**